include/deal.II/grid/reference_cell.h

```diff
@@ -2687,6 +2687,23 @@ namespace internal
      */
     const ArrayView<const T> vertices_1;
   };
+
+  /**
+   * This exception is raised whenever the types of two reference cell objects
+   * were assumed to be equal, but were not.
+   *
+   * Parameters to the constructor are the first and second reference cells,
+   * both of type <tt>ReferenceCell</tt>.
+   */
+  DeclException2(
+    ExcNonMatchingReferenceCellTypes,
+    ReferenceCell,
+    ReferenceCell,
+    << "The reference-cell type of " << arg1.to_string()
+    << " does not match with " << arg2.to_string() << ". "
+    << "This could happen if one object (e.g. Triangulation) holds "
+    << "reference cells from a hexahedral mesh and the other one "
+    << "(e.g. FiniteElement) from a simplex mesh.");
```

Comment:
I still think that this message is too specific. What do you think of the text I had in the other comment?

Reply:
Excellent, thanks, I adopted it. I got so caught up in the other part that I didn't notice the changes in the text initially :-).